html5lib/html5parser.py

```diff
@@ -205,45 +205,43 @@ def mainLoop(self):
             prev_token = None
             new_token = token
             while new_token is not None:
-                prev_token = new_token
-                currentNode = self.tree.openElements[-1] if self.tree.openElements else None
-                currentNodeNamespace = currentNode.namespace if currentNode else None
-                currentNodeName = currentNode.name if currentNode else None
-
                 type = new_token["type"]
-
                 if type == ParseErrorToken:
                     self.parseError(new_token["data"], new_token.get("datavars", {}))
-                    new_token = None
-                else:
-                    if (len(self.tree.openElements) == 0 or
-                        currentNodeNamespace == self.tree.defaultNamespace or
-                        (self.isMathMLTextIntegrationPoint(currentNode) and
-                         ((type == StartTagToken and
-                           token["name"] not in frozenset(["mglyph", "malignmark"])) or
-                          type in (CharactersToken, SpaceCharactersToken))) or
-                        (currentNodeNamespace == namespaces["mathml"] and
-                         currentNodeName == "annotation-xml" and
-                         type == StartTagToken and
-                         token["name"] == "svg") or
-                        (self.isHTMLIntegrationPoint(currentNode) and
-                         type in (StartTagToken, CharactersToken, SpaceCharactersToken))):
+                    break
+
+                prev_token = new_token
```

Comment:
NB: This is technically a behaviour change since `prev_token` could previously have referenced a `ParseErrorToken`.  That said, I don't believe that the [code that inspects the `prev_token`](https://github.com/html5lib/html5lib-python/blob/2c19b9899ab3a3e8bd0ca35e5d78544334204169/html5lib/html5parser.py#L248-L251) would ever be relevant for error tokens.